.github/workflows/scala-new.yml

```
@@ -14,19 +16,24 @@ jobs:
       - ${{ matrix.os }}
       - self-hosted
       - engine
-    timeout-minutes: 120
+    timeout-minutes: 180
     strategy:
       matrix:
+        # The macOS support is not yet done. For now, the GitHub-hosted runner
+        # in the old workflow should be relied upon.
         os: [Windows, Linux]
       fail-fast: false
     steps:
       - uses: actions/checkout@v2
-      - run: cargo run --release --bin build-enso -- ${{ github.workspace }}
-        working-directory: C:\ci
-        if: runner.os == 'Windows'
-      - run: cargo run --release --bin build-enso -- ${{ github.workspace }}
-        working-directory: /ci
-        if: runner.os != 'Windows'
+        with:
+          repository: enso-org/ci-build
+          branch: deploy
```

Are you sure that this `branch` option has any effect? I didn't see it documented at https://github.com/actions/checkout nor in any workflows yet. Maybe you meant `ref`?

Also looking at the https://github.com/enso-org/ci-build repository it seems that the branch there is called `deployed`. 

I'd guess what has been happening for now is it was checking out the default branch.

```suggestion
          ref: deployed
```